include/hipSYCL/sycl/libkernel/id.hpp

```diff
@@ -283,6 +283,45 @@ struct id {
   HIPSYCL_ID_BINARY_OP_SIZE_T(&)
   HIPSYCL_ID_BINARY_OP_SIZE_T(|)
   HIPSYCL_ID_BINARY_OP_SIZE_T(^)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(&&)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(||)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(<)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(>)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(<=)
+  HIPSYCL_ID_BINARY_OP_SIZE_T(>=)
```

Comment:
These operators return `bool`. Do we need a `static_cast` to get it to return a `size_t` without warnings?